src/main/java/base/gameobjects/animalstates/EatingState.java

```diff
@@ -60,16 +72,17 @@ private boolean isNearFood(Game game, Animal animal) {
         }
         for (FoodBowl bowl : game.getGameMap(animal.getCurrentMap()).getFoodBowls()) {
             if (bowl.isFull() && animal.getRectangle().intersects(bowl.getRectangle())) {
+                foodTypeEating = bowl.getFoodType();
```

Comment:
Please assign default value to `foodTypeEating` when animal finds item (food that is not in bowl) (couple of lines earlier)
Otherwise there is a bug in scenario like this:
1. Animal eats perfect meal from the bowl
2. `foodTypeEating` get value 100
3. Animal gets hungry again
4. Animal cannot find food in the bowl and eats random item
5. `foodTypeEating` doesnt change and is still 100
6. saturation amount for eating item is 100 instead of 25